lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -588,6 +588,58 @@ LogicalResult MulOp::canonicalize(MulOp op, PatternRewriter &rewriter) {
   return failure();
 }
 
+template <class Op, bool isSigned>
+static OpFoldResult foldDiv(Op op, ArrayRef<Attribute> constants) {
+  APInt value;
+  if (matchPattern(op.rhs(), m_RConstant(value))) {
```

Comment:
For folders, please use the "constants" thing coming in instead of matchPattern.  This is more powerful for algorithms like SCCP, and is also more efficient.

Canonicalizers should use matchPattern

Reply:
Ok, thank you for pointing out!